featuretools/tests/entityset_tests/test_es.py

```diff
@@ -480,7 +481,7 @@ def test_concat_entitysets(es):
     assert old_es_1.__eq__(es_1, deep=True)
     assert old_es_2.__eq__(es_2, deep=True)
 
-    assert es_3.__eq__(es, deep=True)
+    assert es_3.__eq__(es)
```

Comment:
If this is due to no longer sorting data for entities without time indexes, we could potentially not split up the data of non-time entities and then we could still do a deep comparison

Reply:
below, we check for the fact that the dataframes have the same data after concatenation, so I think it's fine to just remove the deep check here and not mess with the test